### DISCUSSION TITLE
fonts in all environment

### DISCUSSION THREAD

**Discussion by andres-gimenez** [typst/typst#5013] [General]
Is there a way to be able to use otf fonts and have them work in any environment, without having to install them on the machine where typts run?

I need to use the Open Dyslexic fonts and I don't want to have to install the fonts on every machine where I use typts.

When do you use plug-ins in VSCode, where do you need to install the fonts?

**Comment by Enivex** [comment]
The alternative would be bundling the fonts with each project, to make them self contained.

As for vscode: The same way you install fonts for anything else (system)

**Reply by andres-gimenez** [reply]
How are self-contained fonts made?

**Reply by Enivex** [reply]
> How are self-contained fonts made? 

I meant that the project is self contained, since the fonts are included with it

**Reply by andres-gimenez** [reply]
I understood what you mean, but I haven't found any examples nor have I been able to make a reference to the font type, attaching the otf files to the project directory.

**Reply by eike** [reply]
Online, Typst uses font files in the project automatically. Locally, you have to specify the ˋ--font-pathˋ when invoking the compiler. But it should be able to read .otf fonts. This is described here: https://typst.app/docs/reference/text/text/#parameters-font

**Reply by hunkyjimpjorps** [reply]
Since you're using Tinymist in VS Code, if you've got a folder of fonts you want to use, set the path to it using the "Tinymist: Font Paths" option in Settings.

For example, if you want to look for fonts in the same folder as your Typst file and in a fixed fonts folder on your hard drive, you'd set
```
    "tinymist.fontPaths": [
        ".",
		"C:\Projects\Accessible Fonts"
    ]
```

NB: You probably don't want to use `.` if you're compiling Typst files that are in your home directory or some other crowded folder with lots of subfolders and files, since Typst will recursively scan everything in that folder for fonts each time you compile.
